src/components/Movie/MovieDetails.js

```diff
@@ -0,0 +1,62 @@
+import React, {Component} from 'react';
+
+import Constants from '../Common/constants';
+import Header from '../Header/Header';
+import Footer from '../Footer/Footer';
+import SimilarMovies from './SimilarMovies';
+import MovieBanner from './MovieBanner';
+import './moviedetails.css';
+
+export default class MovieDetails extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            movie: {},
+            selectedGenre: '',
+            similarMovies: []
+        };
+    }
+    componentDidMount() {
+        let url = `${Constants.baseURL}/${this.props.match.params.id}`;
+        fetch(url)
+            .then(res => res.json())
+            .then(response => {
+                this.setState({
+                    movie: response
+                });
+                this.fetchSimilarMovies(response.genres[0]);
+            });
+    }
+    componentWillReceiveProps(nextProps) {
+        let url = `${Constants.baseURL}/${nextProps.match.params.id}`;

```

Comment:
Your `componentDidMount` and `componentWillReceiveProps` have basically the same function, but duplicated. What about having it in a separated function that you can call in both places? (DRY)
Other thing, is a good practice checking if the property you want have changed to avoid unnecessary fetch calls.

Reply:
Done